src/core/mac/mac.cpp

```diff
@@ -1274,6 +1274,12 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
         otDumpDebgMac("TX", aFrame.GetHeader(), aFrame.GetLength());
         FinishOperation();
         Get<MeshForwarder>().HandleSentFrame(aFrame, aError);
+#if OPENTHREAD_THREAD_VERSION >= OPENTHREAD_THREAD_VERSION_1_2
+        if (aFrame.GetSecurityEnabled() && aError == OT_ERROR_NONE && aAckFrame != NULL)
+        {
+            Get<DataPollSender>().ResetKeepAliveTimer();
```

Comment:
There needs to be an additional check here so that the timer is only reset when the parent is a Thread 1.2 device. (currently logic is only depending on child [own] version)

Reply:
Thank you! updated.